Makefile

```diff
@@ -2,32 +2,43 @@
 # Synse Emulator Plugin
 #
 
-PLUGIN_NAME := emulator
-PLUGIN_VERSION=1.0
-
-PKG_CTX=main
+PLUGIN_NAME    := emulator
+PLUGIN_VERSION := 1.0.1
+IMAGE_NAME     := vaporio/emulator-plugin
+
+# In CI, git commit is CIRCLE_SHA1 and git tag
+# is CIRCLE_TAG -- perhaps we could consolidate
+# so we prefer those and use this as a fallback?
+GIT_COMMIT ?= $(shell git rev-parse --short HEAD 2> /dev/null || true)
+GIT_TAG    ?= $(shell git describe --tags 2> /dev/null || true)
 BUILD_DATE := $(shell date -u +%Y-%m-%dT%T 2> /dev/null)
-GIT_COMMIT := $(shell git rev-parse --short HEAD 2> /dev/null || true)
-GIT_TAG    := $(shell git describe --tags 2> /dev/null || true)
 GO_VERSION := $(shell go version | awk '{ print $$3 }')
 
-LDFLAGS="-X ${PKG_CTX}.BuildDate=${BUILD_DATE} \
+PKG_CTX    := main
+LDFLAGS	   := -w \
+	-X ${PKG_CTX}.BuildDate=${BUILD_DATE} \
 	-X ${PKG_CTX}.GitCommit=${GIT_COMMIT} \
 	-X ${PKG_CTX}.GitTag=${GIT_TAG} \
 	-X ${PKG_CTX}.GoVersion=${GO_VERSION} \
-	-X ${PKG_CTX}.VersionString=${PLUGIN_VERSION}"
+	-X ${PKG_CTX}.VersionString=${PLUGIN_VERSION}
+
 
 HAS_LINT := $(shell which gometalinter)
 HAS_DEP  := $(shell which dep)
+HAS_GOX  := $(shell which gox)
 
 
+#
+# Local Targets
+#
+
 .PHONY: build
 build:  ## Build the plugin Go binary
-	go build -ldflags ${LDFLAGS} -o build/emulator
+	go build -ldflags "${LDFLAGS}" -o build/emulator
```

Comment:
I would double check make sure this run properly in CI. String escaping is especially sensitive in `ldflags`.

Reply:
It looks like this works:

```
#!/bin/bash -eo pipefail
make build

go build -ldflags "-w -X main.BuildDate=2018-05-11T16:52:00 -X main.GitCommit=5f25594 -X main.GitTag=1.0-6-g5f25594 -X main.GoVersion=go1.10.2 -X main.VersionString=1.0.1" -o build/emulator

```